packages/core/src/define-hook.ts

```diff
@@ -1,4 +1,5 @@
 import type { Hook as HookEntity } from '@workflow/world';
+import type { ZodType } from 'zod';
```

Comment:
We _might_ want to specify `zod` as an optional peer dependency. May be ok to leave out for now though, we could follow up and add that later if necessary.

Reply:
this makes effecttype, arktype, etc. all work well. Will also need to update the docs change to reflect that this is compatible with anything (and then use zod as the example)